packages/react-aria-components/src/Button.tsx

```diff
@@ -156,27 +156,47 @@ export const Button = /*#__PURE__*/ createHideableComponent(function Button(prop
     }
     wasPending.current = isPending;
   }, [isPending, isFocused, ariaLabelledby, buttonId]);
+  let pendingButtonProps: DOMAttributes<HTMLButtonElement> = useMemo(() => isPending ? {
+    onKeyDown: (e) => {
+      if ((e.key === 'Enter' || e.key === ' ') && e.currentTarget instanceof HTMLButtonElement) {
+        e.preventDefault();
+      }
+    },
+    onClick: (e) => {
+      if (e.currentTarget instanceof HTMLButtonElement) {
+        e.preventDefault();
+      }
+    }
+  } : {}, [isPending]);
 
+  // When the button is in a pending state, we want to stop implicit form submission (ie. when the user presses enter on a text input).
+  // We do this by rendering a hidden submit button that is disabled BEFORE the actual submit button as a form's default button is the first submit button
```

Comment:
Wouldn't preventing default in onClick of the regular button handle that? Why do we need a second button?

Reply:
Or change the `type="submit"` to `type="button"`?

Reply:
changing the button type might be a better alternative, but preventing default didn't appear to do anything

Reply:
I'm not quite clear on why a single input will still submit implicitly
I could get the owning form and prevent it from there instead possibly?

Reply:
so is the onKeyDown and onClick handler needed anymore since you changed the type?